compute_tools/src/tls.rs

```diff
@@ -73,15 +75,18 @@ fn try_update_key_path_blocking(pg_data: &Path, tls_config: &TlsConfig) -> Resul
         .create(true)
         .truncate(true)
         .mode(0o600)
-        .open(pg_data.join(SERVER_KEY))?;
+        .open(postgresql_conf_path.join(SERVER_KEY))?;
 
     let mut crt_file = std::fs::OpenOptions::new()
         .write(true)
         .create(true)
         .truncate(true)
         .mode(0o600)
-        .open(pg_data.join(SERVER_CRT))?;
+        .open(postgresql_conf_path.join(SERVER_CRT))?;
 
+    // There's a chance that postgres/pgbouncer/local_proxy reloads halfway between
+    // these writes and reads the wrong keys to the wrong certs.
+    // Not sure how to prevent that.
```

Comment:
Should we add a lock or whatever the async equivalent would be?

Reply:
Doesn't matter. It's postgres that needs the lock

Reply:
Right, I'm saying we add a lock that guarantees that these two code paths don't conflict.

```
let guard = lock.unlock().await;

pg.reload();
```

```
let guard = lock.unlock().await;

// tls stuff
```

Reply:
Not really sufficient since postgres could reload on its own, right?

Reply:
> Not really sufficient since postgres could reload on its own, right?

Maybe I am not understanding what you mean by reload. Do you mean SIGHUP?

Reply:
Could postgres not reload the config on its own?

Reply:
On neon:

```
neondb_owner@neondb=> SELECT pg_reload_conf();
ERROR:  permission denied for function pg_reload_conf
```